src/scale_limits_analyzer.rs

```diff
@@ -0,0 +1,157 @@
+        _scoped_type: bluejay_core::definition::TypeDefinitionReference<
+            '_,
+            <SchemaDefinition<'_> as CoreSchemaDefinition>::TypeDefinition,
+        >,
+        _included: bool,
+    ) {
+        self.path_stack.push(field.response_key());
+        let rate = Self::rate_for_field_definition(field_definition);
+        let values = self.value_stack.last().unwrap();
+        let mut nested_values = Vec::new();
+        values.iter().for_each(|value| {
+            let value_for_field = match value {
+                Value::Object(object) => object.get(field.response_key()),
+                Value::Null => None,
+                _ => panic!("invalid value"),
+            };
+            // TODO handle scaling for this value
+            if let Some(rate) = rate {
+                let length = match value_for_field {
+                    Some(Value::String(s)) => s.len(),
+                    Some(Value::Array(arr)) => arr.len(),
+                    _ => 1,
+                };
+                let increment = length as f64 * rate;
+
+                if let Some(cumulative_rate_for_path) = self.rates.get_mut(&self.path_stack) {
+                    *cumulative_rate_for_path += increment;
+                } else {
+                    self.rates.insert(self.path_stack.clone(), increment);
+                }
+            }
+
+            match value_for_field {
+                Some(Value::Array(values)) => nested_values.extend(values),
+                Some(value) => nested_values.push(value),
+                None => {}
+            }
+        });
+
+        self.value_stack.push(nested_values);
+    }
+
+    fn leave_field(
+        &mut self,
+        _field: &'a <ExecutableDocument<'a> as bluejay_core::executable::ExecutableDocument>::Field,
+        _field_definition: &'a <SchemaDefinition<'a> as CoreSchemaDefinition>::FieldDefinition,
+        _scoped_type: bluejay_core::definition::TypeDefinitionReference<
+            'a,
+            <SchemaDefinition<'a> as CoreSchemaDefinition>::TypeDefinition,
+        >,
+        _included: bool,
+    ) {
+        self.path_stack.pop().unwrap();
+        self.value_stack.pop().unwrap();
+    }
+}
+
+impl<'a>
+    bluejay_validator::executable::operation::Analyzer<
+        'a,
+        ExecutableDocument<'a>,
+        SchemaDefinition<'a>,
+        serde_json::Map<String, serde_json::Value>,
+    > for ScaleLimits<'a>
+{
+    type Output = f64;
+
+    fn into_output(self) -> Self::Output {
+        self.rates
+            .into_values()
+            .fold(Self::MIN_SCALE_FACTOR, f64::max)
+            .clamp(Self::MIN_SCALE_FACTOR, Self::MAX_SCALE_FACTOR)
+    }
+}
+
+impl<'a> ScaleLimits<'a> {
+    const MIN_SCALE_FACTOR: f64 = 1.0;
+    const MAX_SCALE_FACTOR: f64 = 10.0;
+
+    fn rate_for_field_definition(
+        field_definition: &FieldDefinition<DefaultContext>,
+    ) -> Option<f64> {
+        field_definition.directives().and_then(|directives| {
```

Comment:
Nit: You can _probably_ use `.flat_map` to remove some of the indirection but it's not a big deal.